### PR TITLE
N4JC-8 | Added delete node and relationships feature

### PR DESCRIPTION
**Pull request by sandovalezequiel**

### CODE REVIEW

File: src/main/java/org/mule/modules/neo4j/internal/client/Neo4JClientImpl.java [hunk 1]
```diff
@@ -48,8 +48,8 @@ public void updateNodes(String label, Map<String, Object> parameters, Map<String
     }
 
     @Override
-    public void deleteNodes(String label, Map<String, Object> parameters) {
-        execute("MATCH (a:`%s` %s) DELETE a", label, parameters);
+    public void deleteNodes(String label, Boolean removeRelationships, Map<String, Object> parameters) {
+        execute(format("MATCH (a:`%%s` %%s) %s DELETE a",removeRelationships?"DETACH":""), label, parameters);
```
Comment (urbanusjam) [inline src/main/java/org/mule/modules/neo4j/internal/client/Neo4JClientImpl.java]: What happens if `removeRelationships`is null?

File: src/main/java/org/mule/modules/neo4j/internal/connector/Neo4jConnector.java [hunk 2]
```diff
@@ -53,8 +54,8 @@ public void updateNodes(@Default("#[payload]") String label, @Optional @RefOnly
     }
 
     @Processor
-    public void deleteNodes(@Default("#[payload]") String label, @Optional @RefOnly Map<String, Object> parameters) {
-        getClient().deleteNodes(label, parameters);
+    public void deleteNodes(@Default("#[payload]") String label, @FriendlyName("Also remove existing relationships") Boolean removeRelationships, @Optional @RefOnly Map<String, Object> parameters) {
```
Comment (urbanusjam) [inline src/main/java/org/mule/modules/neo4j/internal/connector/Neo4jConnector.java]: "Also remove existing relationships" > "Remove existing relationships"

File: src/main/java/org/mule/modules/neo4j/internal/connector/Neo4jConnector.java [hunk 3]
```diff
@@ -53,8 +54,8 @@ public void updateNodes(@Default("#[payload]") String label, @Optional @RefOnly
...
+        getClient().deleteNodes(label,removeRelationships,parameters);
```
Comment (urbanusjam) [inline src/main/java/org/mule/modules/neo4j/internal/connector/Neo4jConnector.java]: Apply formatter.

File: src/test/java/org/mule/modules/neo4j/automation/functional/DeleteNodesIT.java [hunk 4]
```diff
@@ -16,19 +21,49 @@
...
     public void deleteNodeWithParamsTest() throws JsonProcessingException {
-        assertThat(getTestLabelNode(), equalTo(TOMHANKS_NODE));
-        getConnector().deleteNodes(TEST_LABEL, PARAMS_MAP);
-        assertThat(getTestLabelNode(), equalTo(EMPTY_JSON_LIST));
+        assertThat(getTestLabelNode(TEST_LABEL), equalTo(TOMHANKS_NODE));
+        getConnector().deleteNodes(TEST_LABEL,false, PARAMS_MAP);
+        assertThat(getTestLabelNode(TEST_LABEL), equalTo(EMPTY_JSON_LIST));
+    }
+
+    @Test ( expected =  Neo4JConnectorException.class)
```
Comment (urbanusjam) [inline src/test/java/org/mule/modules/neo4j/automation/functional/DeleteNodesIT.java]: Remove whitespaces.

File: src/test/java/org/mule/modules/neo4j/automation/functional/DeleteNodesIT.java [hunk 5]
```diff
@@ -16,19 +21,49 @@
...
+    public void deleteNodeFailTest() throws JsonProcessingException {
+        getConnector().createNode(TEST_LABEL2, PARAMS_MAP);
```
Comment (urbanusjam) [inline src/test/java/org/mule/modules/neo4j/automation/functional/DeleteNodesIT.java]: `createNode()` method should be in the @Before statement.

Comment (urbanusjam) [inline src/test/java/org/mule/modules/neo4j/automation/functional/DeleteNodesIT.java]: Dismiss the previous comment. I didn't see the other createNode().

File: src/main/java/org/mule/modules/neo4j/internal/connector/Neo4jConnector.java (hunk 2 above)
Comment (sandovalezequiel) [inline src/main/java/org/mule/modules/neo4j/internal/connector/Neo4jConnector.java]: "Add a checkbox named “Also remove existing relationships” and if it’s not checked and the node has any relationship then raise an error" - Marcos Nuñez Cortés

Comment (urbanusjam) [inline src/main/java/org/mule/modules/neo4j/internal/connector/Neo4jConnector.java]: ¬¬

File: src/test/java/org/mule/modules/neo4j/automation/functional/DeleteNodesIT.java [hunk 6]
```diff
@@ -3,32 +3,69 @@
...
-import static org.junit.Assert.assertThat;
-import static org.mule.modules.neo4j.automation.functional.TestDataBuilder.*;
+import com.fasterxml.jackson.core.JsonProcessingException;
 
 public class DeleteNodesIT extends AbstractTestCases {
 
     @Before
     public void setUp() {
         getConnector().createNode(TEST_LABEL, PARAMS_MAP);
+
```
Comment (martinpaoloni) [inline src/test/java/org/mule/modules/neo4j/automation/functional/DeleteNodesIT.java]: Delete this line.

**Pull request merged**